databroker/headersource/core.py

```diff
@@ -515,7 +515,12 @@ def insert_run_start(run_start_col, run_start_cache,
     col = run_start_col
     run_start = dict(time=time, uid=uid, **kwargs)
     apply_to_dict_recursively(run_start, sanitize_np)
-
+    try:
+        if run_start_given_uid(uid, run_start_col,
```

Comment:
True, but it will do so without the extra network call, right? And, in the case of MongoDB, this means it will be checking redundantly.